pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -251,22 +269,256 @@ func (pl *dynamicResources) EventsToRegister() []framework.ClusterEventWithHint
+		return framework.QueueImmediately
+	}
+
+	// Modifications may or may not be relevant. If the entire
+	// status is as before, then something else must have changed
+	// and we don't care. What happens in practice is that the
+	// resource driver adds the finalizer.
+	originalClaim, ok := oldObj.(*resourcev1alpha2.ResourceClaim)
+	if !ok {
+		// Shouldn't happen.
+		pl.logger.Error(nil, "unexpected old object in isSchedulableAfterClaimAddOrUpdate", "obj", oldObj)
+		return framework.QueueAfterBackoff
+	}
+	if apiequality.Semantic.DeepEqual(&originalClaim.Status, &modifiedClaim.Status) {
+		if loggerV := pl.logger.V(7); loggerV.Enabled() {
+			// Log more information.
+			loggerV.Info("claim for pod got modified where the pod doesn't care", "pod", klog.KObj(pod), "claim", klog.KObj(modifiedClaim), "diff", cmp.Diff(originalClaim, modifiedClaim))
+		} else {
+			pl.logger.V(6).Info("claim for pod got modified where the pod doesn't care", "pod", klog.KObj(pod), "claim", klog.KObj(modifiedClaim))
+		}
+		return framework.QueueSkip
+	}
+
+	pl.logger.V(4).Info("status of claim for pod got updated", "pod", klog.KObj(pod), "claim", klog.KObj(modifiedClaim))
+	return framework.QueueImmediately
+}
+
+// isSchedulableAfterPodSchedulingContextChange is invoked whenever a
+// PodSchedulingContext object is updated or deleted. It checks whether that change
+// made a previously unschedulable pod schedulable (updated) or a new attempt
+// is needed to re-create the object (deleted). It errs on the side of letting
+// a pod scheduling attempt happen.
+func (pl *dynamicResources) isSchedulableAfterPodSchedulingContextChange(pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	obj := newObj
+	if obj == nil {
+		// Delete event.
+		obj = oldObj
+	}
+
+	var podScheduling *resourcev1alpha2.PodSchedulingContext
+	switch obj := obj.(type) {
+	case cache.DeletedFinalStateUnknown:
+		p, ok := obj.Obj.(*resourcev1alpha2.PodSchedulingContext)
+		if !ok {
+			// Shouldn't happen.
+			pl.logger.Error(nil, "unexpected deleted object in isSchedulableAfterPodSchedulingChange", "obj", obj.Obj)
+			return framework.QueueAfterBackoff
+		}
+		podScheduling = p
+	case *resourcev1alpha2.PodSchedulingContext:
+		podScheduling = obj
+	default:
+		// Shouldn't happen.
+		pl.logger.Error(nil, "unexpected object in isSchedulableAfterPodSchedulingChange", "obj", obj)
+		return framework.QueueAfterBackoff
+	}
+
+	if podScheduling.Name != pod.Name || podScheduling.Namespace != pod.Namespace {
+		pl.logger.V(7).Info("PodSchedulingContext for unrelated pod got modified", "pod", klog.KObj(pod), "podScheduling", klog.KObj(podScheduling))
+		return framework.QueueSkip
+	}
+
+	// Deleted? That can happen because we ourselves delete the PodSchedulingContext while
+	// working on the pod. This can be ignored.
+	if oldObj != nil && newObj == nil {
+		pl.logger.V(4).Info("PodSchedulingContext for pod got deleted", "pod", klog.KObj(pod))
+		return framework.QueueSkip
+	}
```

Comment:
This deleted event handling can be done around L372 (before converting type).

Reply:
The log message then shouldn't mention the pod. It's simpler, so I moved the check.